### ISSUE TITLE
Type parameters as constraints dosen't work for methods

### ISSUE THREAD

**Issue by hayeah** [Microsoft/TypeScript#7079]
PR https://github.com/Microsoft/TypeScript/pull/5949 implements F-bounded polymorphism. It doesn't seem to be working correctly for methods.

```js
interface Table<T> {
  // For this method `T` is apparently a brand new type parameter.
  insert<T extends U, U>(obj: U): T;
  // insert(obj: any)
}

interface Foo {
  a: string;
  b: string;
}

let Foos: Table<Foo>;

// `foo1`'s type is inferred to be {a: 1}
let foo1 = Foos.insert({a: 1}) // => ;

// Should produce error
let foo2 = Foos.insert({c: 1});

```

**Comment by ahejlsberg** [comment]
This is by design. As you mention in your comment, your `insert` method introduces a *new* (and unused) type parameter called `T` that is constrained to extend `U`, but there is no constraint on `U` itself. I'm guessing you're trying to express that `U` must be a supertype of `T`. That's not currently possible, but there is some discussion of that issue in #6613.
